### ISSUE TITLE
Automated label scaling for axis_geo()

### ISSUE THREAD

**Issue by willgearty** [palaeoverse/palaeoverse#21]
In `deeptime::coord_geo()`, there is the option to use `ggfittext` to scale the labels such that they don't overlap with one another and fit within their boxes. I'm not aware of something similar for base R, but we could try to emulate it for `axis_geo()`. @KEichenseer looked into it during development but couldn't figure out a solution, so it might not even be possible in base R?

**Comment by LewisAJones** [comment]
Is this likely to be implemented going forward @willgearty? Can we assign this to milestone 1.3.0?

**Comment by willgearty** [comment]
It looks like `ggfittext` does most of its work in `grid`, so we likely could use similar code for base R plots, e.g.:
https://github.com/wilkox/ggfittext/blob/master/R/geom_fit_text.R#L624
A quick glance at the code makes it seem fairly complicated, but I suppose we could add it to the 1.3.0 milestone for now.

**Comment by LewisAJones** [comment]
Do you think you'll have chance to look into this again this autumn @willgearty, or should we aim for this in version 1.5.0?

**Comment by willgearty** [comment]
I don't think this will be an easy addition, so I'd say punt it to the next release.

**Comment by willgearty** [comment]
I thought about this more, and I just don't think this is possible in base R. I don't think there is an easy way to identify how large the box will be in the plot and then figure out how large the font should be.

**Comment by willgearty** [comment]
Hmm...after thinking about this even more, I think that this _may_ be possible through a combination of graphics::strwidth and graphics::grconvertX/Y. I'll take a look sometime.
